data-structures.md

```diff
@@ -43,6 +43,10 @@ type Block struct {
 	// Height is the chain height of this block.
 	Height Uint64
 
+	// BlockSig is a signature over the hash of the entire block with the miners
+	// worker key to ensure that it is not tampered with after creation
+	BlockSig Signature
```

Comment:
Given that the signature can't sign itself, it would be good to maybe put it at the end and say it signs everything before? Or how exactly do you imagine this to work?

Reply:
oh yeah, it should go at the end. 👍